ipaserver/install/cainstance.py

```diff
@@ -37,6 +37,14 @@
 from configparser import RawConfigParser
 from pkg_resources import parse_version
 
+try:
+    # This fn is available in 389-ds 2.5+. Previous versions use bdb as
+    # the database backend.
+    from lib389.utils import get_default_db_lib
+    ldap_backend = get_default_db_lib()
```

Comment:
I'm ok with this method but I have a reservation here: the default method is not necessary the one used by the existing deployment. Worse, for deploying IPA we allow to pass 389-ds settings which might affect choice of the backend via `db_lib` option.

On the other hand, we already have used `get_default_db_lib()` in `DsInstance.__common_setup()` since https://pagure.io/freeipa/issue/9435 has been fixed...

Reply:
I took inspiration from that change.

You raise a good point. By the time we go about installing the CA 389 will always be configured. I can instead use value of nsslapd-backend-implement from cn=config,cn=ldbm database,cn=plugins,cn=config to make this decision so we'll use the actual backend and not only the default.